### DISCUSSION TITLE
[API Request] ReplayGain control for jukebox mode

### DISCUSSION THREAD

**Discussion by dweymouth** [opensubsonic/open-subsonic-api#76] [API requests]
### Type of change

API extension

### Proposal description

I propose adding an OpenSubsonic extension consisting of a single new endpoint to control ReplayGain settings on the server jukebox mode. Perhaps `jukeboxControlReplayGain`? There could be up to 4 properties I can think of, we should discuss to see which ones could be included and supported by the servers:

* **mode**: one of "off", "track", or "album"
* **preamp**: a floating point number that is the preamp gain to apply in dB when replayGain is on. (used because replay gain usually makes songs quieter - some users prefer to offset this by applying a positive gain when replayGain is in effect)
* **preventClipping**: a boolean; if true the replaygain volume should be reduced if necessary to prevent clipping. (ie if the replayGain for a track is positive but the peak is near 1.000, the full positive gain will not be applied but attenuated just enough to prevent clipping)
* **fallbackGain**: default gain to apply for files that do not have ReplayGain tags

I would think we should support at least the first 2; the others are more of a nice-to-have

### Backward compatibility impact

None. New extension

### Backward compatibility

- [x] No backward compatibility impact.

### API details

See proposal description

**Comment by deluan** [comment]
IMO, this should be a new action in the `jukeboxControl` endpoint, not a new endpoint.

**Reply by dweymouth** [reply]
Yeah, that works for me and I don't know why I didn't think of this the first time around
